pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -428,11 +428,22 @@ func (g *DefaultValidator) validateMetricsCollector(inst *experimentsv1beta1.Exp
 			mcSpec.Source.FileSystemPath.Kind != commonapiv1beta1.FileKind || !filepath.IsAbs(mcSpec.Source.FileSystemPath.Path) {
 			return fmt.Errorf("file path where metrics file exists is required by .spec.metricsCollectorSpec.source.fileSystemPath.path")
 		}
+		// FileSystemFileFormat
+		fileFormat := mcSpec.Source.FileSystemPath.FileFormat
+		if fileFormat != commonapiv1beta1.TextFormat && fileFormat != commonapiv1beta1.JsonFormat {
+			return fmt.Errorf("format of metrics file is required by .spec.metricsCollectorSpec.source.fileSystemPath.fileFormat")
+		}
+		if fileFormat == commonapiv1beta1.JsonFormat && mcSpec.Source.Filter != nil {
+			return fmt.Errorf(".spec.metricsCollectorSpec.source.filter must be nil when format of metrics file is `JSON`")
```

Comment:
Let's use `commonapiv1beta1.JsonFormat` instead of `'JSON'` here.

Reply:
It makes sense.

Reply:
Done.